src/dispersion_relation.jl

```diff
@@ -0,0 +1,93 @@
+@doc raw"""
+    LinearDispersionRelation(ref_height)
+
+A struct representing a linear dispersion relation ``\omega(k)`` of an equation. The reference
+water height `h0` is given by `ref_height`.
```

Comment:
```suggestion
water height `h0` is given by `ref_height`. The linear dispersion relation is
used by [`wave_speed`](@ref).
```

Reply:
I added a similar sentence in [9a62605](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/168/commits/9a62605527924b2d5b9e975fa80b3f36e78afee4) and [89324d5](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/168/commits/89324d561c996d5c3af3cbd66ad07e7410619c2b).